pkg/xds/envoy/types.go

```diff
@@ -21,11 +23,118 @@ type ClusterSubset struct {
 }
 
 type Tags map[string]string
+type TagsSlice []Tags
+type TagKeys []string
+type TagKeysSlice []TagKeys
 
-func (t Tags) WithoutTag(tag string) Tags {
+func (t TagsSlice) ToTagKeySlice() TagKeysSlice {
+	out := []TagKeys{}
+	for _, v := range t {
+		out = append(out, v.Keys())
+	}
+	return out
+}
+
+// Transform applies each transformer to each TagSlice and return a sorted unique TagKeysSlice.
+func (t TagKeysSlice) Transform(transformers ...TagKeyTransformer) TagKeysSlice {
+	allSlices := map[uint64]TagKeys{}
+	for _, slice := range t {
```

Comment:
I guess this variable should be named `tagKeys` or `keys` rather than `slice`, right?